README.md

```diff
@@ -502,6 +511,115 @@ server:start()
 
 [socket_ref]: https://www.tarantool.io/en/doc/latest/reference/reference_lua/socket/#socket-tcp-server
 
+## Roles
+
+New roles could be accessed from this project.
+
+### `roles.httpd`
+
+This role can be used only with Tarantool 3. It allows configuring one or more
+HTTP servers. Those servers could be reused by several other roles.
+
+Example of the configuration:
+
+```yaml
+roles_cfg:
+  roles.httpd:
+    default:
+    - listen: 8081
+    additional:
+    - listen: '127.0.0.1:8082'
+```
+
+Server address should be provided either as a URI or as a single port
+(in this case, `0.0.0.0` address is used).
+
+User can access every working HTTP server from the configuration by name,
+using `require('roles.httpd').get_server(name)` method.
+If the `name` argument is `nil`, the default server is returned
+(its name should be equal to constant
+`require('roles.httpd').DEFAULT_SERVER_NAME`, which is `"default"`).
+
+Let's look at the example of using this role. Consider a new role
+`roles/hello_world.lua`:
+```lua
+local M = {}
+local server = {}
+
+M.validate = function()  end
+
+M.apply = function(conf)
+    server = require('roles.httpd').get_server(conf.httpd).httpd
+
+    server:route({
+        path = '/hello/world',
+        name = 'greeting',
+    }, function(tx)
+        return tx:render({text = 'Hello, world!'})
+    end)
+end
+
+M.stop = function()
+    server:delete('greeting')
+end
+
+return M
+```
+
+This role accepts a server by name from a config and creates a route to return 
+`Hello, world!` to every request by this route.
+
+Then we need to write a simple config to start the Tarantool instance via
+`tt`:
+```yaml
+app:
+  file: 'myapp.lua'
+
+groups:
+  group001:
+    replicasets:
+      replicaset001:
+        roles: [roles.httpd, roles.hello_world]
+        roles_cfg:
+          roles.httpd:
+            default:
+              listen: 8081
+            additional:
+              listen: '127.0.0.1:8082'
+          roles.hello_world:
+            httpd: 'additional'
+        instances:
+          instance001:
+            iproto:
+              listen:
+                - uri: '127.0.0.1:3301'
+```
+
+In `myapp.lua` we need to write a script to initialize both roles:
+```lua
+local httpd_role = require('roles.httpd')
+local hello_role = require('roles.hello_world')
+
+httpd_role.apply(config.roles_cfg.httpd)
+hello_role.apply(config.roles_cfg.hello_world)
+```
+
```

Comment:
It could be removed, just config.yaml + tt start should be enough.

Reply:
Yes, `role.apply` is not designed to be called directly in applications/other roles. (We may do so in unit tests, but just for the sake of unit testing.)